src/bin/nydus-image/core/node.rs

```diff
@@ -999,7 +1001,15 @@ impl Node {
             return false;
         }
 
-        self.inode.is_chrdev() && stat::major(self.rdev) == 0 && stat::minor(self.rdev) == 0
+        #[cfg(target_os = "linux")]
+        {
+            use nix::sys::stat;
+            self.inode.is_chrdev() && stat::major(self.rdev) == 0 && stat::minor(self.rdev) == 0
+        }
+        #[cfg(target_os = "macos")]
+        {
+            self.inode.is_chrdev() && ((self.rdev >> 24) & 0xff) == 0 && (self.rdev & 0xffffff) == 0
```

Comment:
please add two more helpers `major_dev` and `minor_dev` to wrap around the differences between macos and linux.